devel/mercurial/Portfile

```diff
@@ -39,12 +39,11 @@ long_description    Mercurial is a fast, lightweight Source Control Management \
 homepage            https://www.mercurial-scm.org
 
 checksums           ${distname}${extract.suffix}  \
-                    rmd160  c241985c19032ed79dc170457a70f37677ccce3a \
-                    sha256  f75d6a4a75823a1b7d713a4967eca2f596f466e58fc6bc06d72642932fd7e307 \
-                    size    8253219
+                    rmd160  f92d38d1e41dbc8280b4bee90cc9638dcf49553e \
+                    sha256  5358ed20604c67a7bc7ff81141f735586c328fbb6a18440d2e1e6cd0c044a4c6 \
+                    size    8300870
 
-# keep 3.11 for now, as 3.12 breaks the rust variant
-python.default_version 311
+python.default_version 312
```

Comment:
Did you test the `+rust` variant? I believe this breaks it.

Reply:
It builds fine. Any other testing needs to be done?

Reply:
> It builds fine. Any other testing needs to be done?

Yes, you also need to verify that it runs. Last I tested it, the Rust bindings didn't support 3.12, although that was prior to the release. As noted, you also need to update other ports, such as TortoiseHg, hg-git and hg-evolve.

Reply:
> > It builds fine. Any other testing needs to be done?
> 
> Yes, you also need to verify that it runs. Last I tested it, the Rust bindings didn't support 3.12, although that was prior to the release. As noted, you also need to update other ports, such as TortoiseHg, hg-git and hg-evolve.

Just tried and got "symbol not found in flat namespace" error - I assume this is what you mean by "Rust bindings didn't support 3.12"?

Reply:
I can add in python variants and makes it error out on `+python312 +rust`, if you're ok with it. It will still default to using 3.11 if no variant is specified.

Reply:
adding variants here makes no sense, please don't do that. Just update it to Python 3.12 once that works.

Reply:
Cool. I'll be reverting python related changes.